### PR TITLE
Fix incorrect last test date in Test Questionnaire

### LINKED ISSUE #2130: [BUG] Previous test dates in the Test questionnaire are all displaying as 12:00am
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In the Test Questionnaire, the date for a last test in the system is displaying as 12:00am regardless of the actual time at which the test was conducted.

### Steps To Reproduce

1. Enter a test for a person with a specified date and a time that is not 12:00AM
2. Submit the test
3. Go to the Results page
4. Verify the test date/time is correctly recorded
5. Go to the Conduct Test page
6. Create a test for the same person
7. Note the previous test displays as having been conducted at 12:00AM

### Expected Behavior

The correct time shows for the previous test

### Additional context

image (3)

### Does this issue have a priority label?

- [X] I have added a priority label

### Has issue been assigned to a team?

- [X] I have added a team label

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- Fixes #2130

## Changes Proposed

- Fixes a display bug caused by accidentally stripping off the time from a datetime string a little too early
- Removed test that was asserting the wrong thing
- Added a snapshot test to catch regressions

## Screenshots / Demos

displaying correct time

### UI
- [n/a] Any changes to the UI/UX are approved by design 
- [n/a] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [n/a] Database changes are submitted as a separate PR
  - [n/a] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [n/a] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [n/a] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- [n/a] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [n/a] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [n/a] Any dependencies introduced have been vetted and discussed

## Cloud
- [n/a] DevOps team has been notified if PR requires ops support
- [n/a] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: frontend/src/app/testQueue/AoEForm/AoEForm.test.tsx [hunk 1]
```diff
@@ -115,9 +115,7 @@ describe("AoEForm", () => {
       </MockedProvider>
     );
 
-    expect(component.root.props.children.props.lastTest.dateTested).toEqual(
```
Comment (nullflux) [inline frontend/src/app/testQueue/AoEForm/AoEForm.test.tsx:118]: Thanks for fixing this 😬

File: frontend/src/app/testQueue/AoEForm/AoEForm.tsx [hunk 2]
```diff
@@ -139,11 +139,6 @@ const AoEForm: React.FC<Props> = ({
 
   const patientIsOver18 = moment().diff(patient.birthDate, "years") >= 18;
 
-  // Null is OK and preferred over an empty string
-  if (lastTest?.dateTested) {
```
Comment (nullflux) [inline frontend/src/app/testQueue/AoEForm/AoEForm.tsx:143]: **question:** So is the idea now that `lastTest.dateTested` is always present and correctly formatted?

**Review by nullflux — APPROVED** [review]
Provisionally approved w/ a clarifying question!

File: frontend/src/app/testQueue/AoEForm/AoEForm.tsx (hunk 2 above)
Comment (nickscialli-usds) [reply frontend/src/app/testQueue/AoEForm/AoEForm.tsx:143]: It's always coming from the backend, so it should be consistent. Furthermore, if you look at how `lastTest.dateTested` is used (it's only used in `PriorTestInputs.tsx`), it's either only used as a truthy/falsey value or it's passed to a formatter

**Pull request merged**